### DISCUSSION TITLE
Philips Hue integration

### DISCUSSION THREAD

**Discussion by laurentkempe** [AutoDarkMode/Windows-Auto-Night-Mode#534] [Custom scripts]
Integrate the control of [Philips Hue lighting](https://www.philips-hue.com/en-us) to adapt the lightning of a room.

An alternative would be to be able to run an external tool when events are raised.

**Comment by Spiritreader** [comment]
That seems like a very good idea!
However, in order to implement this, we would need someone who owns and uses Phillips Hue to submit a [Component pull request](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/wiki/Adding-a-new-component), or alternatively, eventually uses the scripting engine that is in planning (#186) once available.

For us maintainers, this is currently out of scope, as we do not possess the necessary hardware to test it.

**Comment by laurentkempe** [comment]
Thanks for your answer @Spiritreader
It looks like the scripting engine would work and I could have a look to the integration when it will be available.

**Comment by Spiritreader** [comment]
Custom scripting is now available in the latest beta builds.
For references, see: https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/wiki/How-to-add-custom-scripts

**Comment by eymay** [comment]
Open to test scripts on Hue's!

In terms of scenarios it should be possible to check wifi for location and bluetooth for proximity to the Hue bulb. Or a connection to Home Assistant could be amazing.

**Comment by JimmyRespawn** [comment]
Another idea is to connect to the entity of HASS. Like the lightness. If value > xxx => light. value < xxx => dark
